beacon_node/store/src/hot_cold_store.rs

```diff
@@ -43,6 +43,63 @@ use types::data_column_sidecar::{ColumnIndex, DataColumnSidecar, DataColumnSidec
 use types::*;
 use zstd::{Decoder, Encoder};
 
+#[derive(Debug, Clone)]
+pub enum BlobsSidecarListFromRoot<E: EthSpec> {
+    /// Valid root that exists in the DB, but has no blobs associated with it.
+    NoBlobs,
+    /// Contains > 1 blob for the requested root.
+    Blobs(BlobSidecarList<E>),
+    /// No root exists in the db or cache for the requested root.
+    NoRoot,
+}
+
+impl<E: EthSpec> From<BlobSidecarList<E>> for BlobsSidecarListFromRoot<E> {
+    fn from(value: BlobSidecarList<E>) -> Self {
+        Self::Blobs(value)
+    }
+}
+
+impl<E: EthSpec> Encode for BlobsSidecarListFromRoot<E> {
+    fn as_ssz_bytes(&self) -> Vec<u8> {
+        match self {
+            Self::NoBlobs | Self::NoRoot => vec![],
```

Comment:
I think this is right, but can we add a test that asserts `vec![].as_ssz_bytes()` and `BlobSidecarListFromRoot::NoBlobs.as_ssz_bytes()` are equal? And maybe similarly for a vec of vecs, and a vec of `BlobSidecarListFromRoot` so we exercise the `ssz_append` impl as well